### PR TITLE
Allow null in Boolean.convertTo(True|False) functions (#9146).

### PR DESCRIPTION
**Pull request by sgpinkus**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory.

If you're making a change to documentation, do **not** modify a `.html` file directly. Instead find the corresponding `.pug` file or test case in the `test/docs` directory.

**Summary**
Allow Boolean.convertToFalse Set to take null. See [comment](https://github.com/Automattic/mongoose/issues/9146#issuecomment-656084916) on #9146 for motivation.

**Examples**

```
const mongoose = require('/home/sam/src/mongoose/index.js');
const { Schema } = mongoose;
mongoose.Schema.Types.Boolean.convertToFalse.add(null);
console.log(mongoose.Schema.Types.Boolean.convertToFalse);
const M = mongoose.model('Test', new Schema({ b: Boolean }));
console.log(new M({ b: null })); // false
```

### CODE REVIEW

File: lib/schema/boolean.js [hunk 1]
```diff
@@ -224,6 +224,8 @@ SchemaBoolean.prototype.castForQuery = function($conditional, val) {
   return this._castForQuery($conditional);
 };
 
+SchemaBoolean.prototype.castNull = true;
```
Comment (vkarpov15) [inline lib/schema/boolean.js:227]: Is this change necessary given your other changes? Your code example doesn't make any reference to this new `castNull` option.

Comment (AbdelrahmanHafez) [reply lib/schema/boolean.js:227]: It's being used in [this line](https://github.com/Automattic/mongoose/blob/18296bb5d6fdc9975e2825dd4fee995fe340c516/lib/schematype.js#L1058).

Comment (sgpinkus) [reply lib/schema/boolean.js:227]: I just used it because it's more extensible than say type checking for Boolean in the line @AbdelrahmanHafez referred to. If in future, other types want to cast null, they'd set `castNull = true` to get passed that conditional check, although currently other type cast implementations (like ex. [Number](https://github.com/Automattic/mongoose/blob/43288f869366ade4c287b4dd06e9c7016c6104c1/lib/cast/number.js#L17)) don't do anything with null even if castNull is true. Boolean is a bit of a special case because there is an existing documented mechanism (convertToTrue|False) exposed to easily customize null casting.

Comment (AbdelrahmanHafez) [reply lib/schema/boolean.js:227]: If this is approved, it would be great if you add this option to the documentation using JSDoc.

**Comment by sgpinkus** [comment]
> I made an alternative fix in [415ba3d](https://github.com/Automattic/mongoose/commit/415ba3de0486727209f6c2ffc9d6edd360235fc2) . Very similar, just no need for the extra `castNull` option. If you add `null` to `convertToFalse`, that should be all you have to do.

Awesome thanks! Yeah, castNull wasn't ideal.
